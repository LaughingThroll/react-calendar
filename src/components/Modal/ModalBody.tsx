import React from "react"
import { TChildren } from "../../types/utilsTypes"

interface IModalBody {
  children?: TChildren
}

const ModalBody: React.FC<IModalBody> = ({ children }: IModalBody) => {
  return (
    <form className="modal-body">
      {children && !Array.isArray(children) && <div className="modal-body__item ">{children}</div>}
      {children &&
        Array.isArray(children) &&
        children.map((child, i) => (
          <div key={i} className="modal-body__item">
            {child}
          </div>
        ))}
    </form>
  )
}

export default ModalBody
