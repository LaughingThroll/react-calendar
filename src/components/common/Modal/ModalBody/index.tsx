import React from 'react'
import styles from './modalBody.module.scss'

interface ModalBodyProps {
  children?: any
}
const { modalBody, modalBody__item } = styles

const ModalBody: React.FC<ModalBodyProps> = ({ children }) => {
  return (
    <form className={modalBody}>
      {React.Children.map(children, (child, i) => (
        <div key={i} className={modalBody__item}>
          {child}
        </div>
      ))}
    </form>
  )
}

export default React.memo(ModalBody)
