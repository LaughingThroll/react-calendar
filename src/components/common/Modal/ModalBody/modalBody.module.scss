.modalBody {
  padding: 15px 27px 30px 38px;
}
