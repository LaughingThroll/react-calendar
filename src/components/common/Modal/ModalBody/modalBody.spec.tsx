import React from 'react'
import { render } from 'enzyme'
import ModalBody from './'

describe('ModalBody render', () => {
  it('render default', () => {
    const tree = render(<ModalBody />)

    expect(tree).toMatchSnapshot()
    expect(tree.find('div')).toHaveLength(0)
  })

  it('render with one child', () => {
    const tree = render(<ModalBody children={<span>Test</span>} />)

    expect(tree).toMatchSnapshot()
    expect(tree.find('div')).toHaveLength(1)
  })

  it('render with children', () => {
    const tree = render(
      <ModalBody children={[<span key="1">Test</span>, <span key="2">Test</span>, <span key="3">Test</span>]} />
    )

    expect(tree).toMatchSnapshot()
    expect(tree.find('div')).toHaveLength(3)
  })
})
