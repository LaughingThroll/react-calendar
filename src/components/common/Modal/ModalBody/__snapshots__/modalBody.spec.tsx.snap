// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBody render render default 1`] = `
<form
  class="modalBody"
/>
`;

exports[`ModalBody render render with children 1`] = `
<form
  class="modalBody"
>
  <div
    class="modalBody__item"
  >
    <span>
      Test
    </span>
  </div>
  <div
    class="modalBody__item"
  >
    <span>
      Test
    </span>
  </div>
  <div
    class="modalBody__item"
  >
    <span>
      Test
    </span>
  </div>
</form>
`;

exports[`ModalBody render render with one child 1`] = `
<form
  class="modalBody"
>
  <div
    class="modalBody__item"
  >
    <span>
      Test
    </span>
  </div>
</form>
`;
